### ISSUE TITLE
Need someway to feed labels into layout spec

### ISSUE THREAD

**Issue by hadley** [thomasp85/ggraph#32]
Ideally here the arrows would touch the closest side of the label.

``` r
library(igraph)
#> 
#> Attaching package: 'igraph'
#> The following objects are masked from 'package:stats':
#> 
#>     decompose, spectrum
#> The following object is masked from 'package:base':
#> 
#>     union
library(ggraph)
#> Loading required package: ggplot2

gr <- make_graph('bull')
E(gr)$class <- sample(letters[1:3], gsize(gr), replace = TRUE)
V(gr)$class <- sample(letters[1:3], gorder(gr), replace = TRUE)

ggraph(gr, 'igraph', algorithm = 'nicely') +
  geom_edge_link0(arrow = grid::arrow(length = unit(0.5, "cm"), type = "closed")) + 
  geom_node_label(aes(label = class), repel = TRUE, label.padding = unit(1, "lines"))
```

![](http://i.imgur.com/7gIEWP0.png)

**Comment by thomasp85** [comment]
Related to #20

I would like to solve this more generally and due to the different types of edges I don't think moving the end point directly will cut it... I do have some ideas for a responsive path grob that I need to look into...

All in all I really don't like arrowheads due to this as well as the clutter they introduce, but for very simple graphs they have their merit so they should get better support...

**Comment by thomasp85** [comment]
I think your use case has been fixed with #35 and using the `label_rect` helper?
